pkg/plugin/implementation/router/router.go

```
@@ -0,0 +1,204 @@
+	if err := yaml.Unmarshal(data, &config); err != nil {
+		return fmt.Errorf("error parsing YAML: %w", err)
+	}
+
+	// Validate rules
+	if err := validateRules(config.RoutingRules); err != nil {
+		return fmt.Errorf("invalid routing rules: %w", err)
+	}
+	r.rules = config.RoutingRules
+	return nil
+}
+
+// validateRules performs basic validation on the loaded routing rules.
+func validateRules(rules []routingRule) error {
+	for _, rule := range rules {
+		// Ensure domain, version, and routingType are present
+		if rule.Domain == "" || rule.Version == "" || rule.RoutingType == "" {
+			return fmt.Errorf("invalid rule: domain, version, and routingType are required")
+		}
+
+		// Validate based on routingType
+		switch rule.RoutingType {
+		case "url":
+			if rule.Target.URL == "" {
+				return fmt.Errorf("invalid rule: url is required for routingType 'url'")
+			}
+			if _, err := url.ParseRequestURI(rule.Target.URL); err != nil {
+				return fmt.Errorf("invalid URL in rule: %w", err)
+			}
+		case "msgq":
+			if rule.Target.TopicID == "" {
+				return fmt.Errorf("invalid rule: topicId is required for routingType 'msgq'")
+			}
+		case "bpp", "bap":
+			// No target validation needed for bpp/bap, as they use URIs from the request body
+			continue
+		default:
+			return fmt.Errorf("invalid rule: unknown routingType '%s'", rule.RoutingType)
+		}
+	}
+	return nil
+}
+
+// Route determines the routing destination based on the request context.
+func (r *Router) Route(ctx context.Context, url *url.URL, body []byte) (*definition.Route, error) {
+
+	// Parse the body to extract domain and version
+	var requestBody struct {
+		Context struct {
+			Domain  string `json:"domain"`
+			Version string `json:"version"`
+			BppURI  string `json:"bpp_uri,omitempty"`
+			BapURI  string `json:"bap_uri,omitempty"`
+		} `json:"context"`
+	}
+	if err := json.Unmarshal(body, &requestBody); err != nil {
+		return nil, fmt.Errorf("error parsing request body: %w", err)
```

this should return model.BadReq

we should return a badRequest error from this PR https://github.com/beckn/beckn-onix/pull/437/files#

---

Will take up as part of integration task.